content/blog/2018-09-04-birds-taxo-traits.md

```diff
@@ -0,0 +1,559 @@
+---
+title: "What are these birds? Complement occurrence data with taxonomy and traits information"
+slug: birds-taxo-traits
+authors:
+  - name: Maëlle Salmon
+    url: https://masalmon.eu/
+date: 2018-09-04
+preface: The blog post series corresponds to the material for a talk Maëlle will give at the [Animal Movement Analysis summer school in Radolfzell, Germany on September the 12th](http://animove.org/animove-2019-evening-keynotes/), in a Max Planck Institute of Ornithology.
+tags:
+- auk
+- taxize
+- traits
+- taxonomy
+- rphylopic
+- gifski
+- birder
+output: 
+  md_document:
+    variant: markdown_github
+    preserve_yaml: true
+---
+
+Thanks to the [second post of the series where we obtained data from
+eBird](https://ropensci.org/blog/2018/08/21/birds-radolfzell/) we know
+what birds were observed in the county of Constance. Now, not all
+species’ names mean a lot to me, and even if they did, there are a lot
+of them. In this post, we shall use rOpenSci’s packages accessing
+taxonomy and trait data in order to summarize the birds’ population of
+the county: armed with scientific and common names of birds, we have
+access to plenty of open data!
+
+### Getting and filtering the occurrence data
+
+For more details about the following code, refer to the [previous post
+of the series](https://ropensci.org/blog/2018/08/21/birds-radolfzell/).
+We only add a step to keep only data for the most recent years.
+
+``` r
+# polygon for filtering
+landkreis_konstanz <- osmdata::getbb("Landkreis Konstanz",
+                             format_out = "sf_polygon")
+crs <- sf::st_crs(landkreis_konstanz)
+
+# get and filter data
+f_out_ebd <- "ebird/ebd_lk_konstanz.txt"
+
+library("magrittr")
+
+ebd <- auk::read_ebd(f_out_ebd) %>%
+  sf::st_as_sf(coords = c("longitude", "latitude"), 
+                crs = crs) 
+
+in_indices <- sf::st_within(ebd, landkreis_konstanz)
+
+ebd <- dplyr::filter(ebd, lengths(in_indices) > 0)
+
+ebd <- as.data.frame(ebd)
+
+ebd <- dplyr::filter(ebd, approved, lubridate::year(observation_date) > 2010)
+
+nrow(ebd)
+```
+
+    ## [1] 8599
+
+We’ll also need these two data.frames later: abundance by species, and
+dictionary of names.
+
+``` r
+abundance <- dplyr::count(ebd, scientific_name)
+
+dico <- unique(dplyr::select(ebd, scientific_name,
+                             common_name))
+```
+
+### Getting taxonomic information
+
+In this section I would like to get an idea of how much of the bird
```

Comment:
in "how much of the bird diversity" sounds specific

Do you mean "how much bird diversity" or "how diverse the types of birds are"?

Reply:
ok "how diverse the types of birds are"